### DISCUSSION TITLE
Isolated custom theming with Autocompletion (Without TypeScript)

### DISCUSSION THREAD

**Discussion by vipindigiqt** [mantinedev/mantine#1888] [Feature requests]
![image](https://user-images.githubusercontent.com/81612126/181171282-313ce628-12bc-4021-bc74-ced6afb7639c.png)

Right now, we only get autocompletions when customizing the theme directly on `MantineProvider`

Can we have a function, through which we can get nice autocompletions and isolate the custom theming from our root file.

Is it somehow possible without TypeScript?

Thanks a lot for **Mantine**, I'm lovin' it.

**Comment by rtivital** [comment]
I think that it is possible only with TypeScript.
